staging/src/k8s.io/apiserver/pkg/cel/common/values_test.go

```diff
@@ -291,10 +299,50 @@ func TestToValue(t *testing.T) {
 			activation: map[string]typedValue{"c": complex1},
 		},
 		{
-			name:       "struct: embedded struct: omitempty struct field",
-			expression: "!has(c.metadata.labels)",
+			name:       "struct: omitempty: zero valued: has struct field",
+			expression: "has(c.nestedEmpty)",
+			activation: map[string]typedValue{"c": complex2},
+		},
+		{
+			name:       "struct: omitempty: zero valued: does not have scalar fields",
+			expression: "!has(c.s) && !has(c.i) && !has(c.b) && !has(c.f)",
+			activation: map[string]typedValue{"c": structOmitEmpty1},
+		},
+		{
+			name:       "struct: omitempty: zero valued: does not have pointer to scalar fields",
+			expression: "!has(c.sp) && !has(c.ip) && !has(c.bp) && !has(c.fp)",
+			activation: map[string]typedValue{"c": structOmitEmpty1},
+		},
+		{
+			name:       "struct: omitzero: zero valued: does not have struct field",
```

Comment:
Is this test case for omitzero? should it be using `"c": structOmitZero1`?

Reply:
This is correct.  (`NestedZero  Nested `json:"nestedZero,omitzero"``)